apps/condo/domains/ticket/components/BaseTicketForm/TicketAssignments.tsx

```diff
@@ -0,0 +1,181 @@
+
+    const [divisions, setDivisions] = useState([])
+
+    const formatUserFieldLabel = ({ text, value, data: employee }) => {
+        if (!employee) {
+            return null
+        }
+        const matchedSpecialization = find(employee.specializations, { id: categoryClassifier })
+        return (
+            <UserNameField user={{ name: text, id: value }}>
+                {({ name, postfix }) => (
+                    <>
+                        <Typography.Text>
+                            {name} {postfix}
+                        </Typography.Text>
+                        {matchedSpecialization && (
+                            <Typography.Text type="secondary">
+                                {`(${matchedSpecialization.name.toLowerCase()})`}
+                            </Typography.Text>
+                        )}
+                    </>
+                )}
+            </UserNameField>
+        )
+    }
+
+    const getTechniciansFrom = (division) => (
+        division.executors.filter(({ specializations }) => (
+            specializations.some(({ id }) => id === categoryClassifier)
+        ))
+    )
+
+    const convertToOption = (employee) => ({
+        text: employee.name,
+        value: employee.user.id,
+        data: employee,
+    })
+
+    /**
+     * Employees are grouped by following rules:
+     * 1. Technicians with matched specialization, belonging to matched division;
+     * 2. Technicians with matched specialization, belonging to other matched divisions;
+     * 3. Rest of employees.
+     */
+    const renderOptionGroups = (employeeOptions, renderOption) => {
+        const [currentDivision, ...otherDivisions] = divisions
+        let techniciansOnDivisionOptions = []
+        let techniciansOnOtherDivisionsOptions = []
+        let otherTechniciansOptions = []
+
+        if (currentDivision) {
+            const techniciansOnDivision = getTechniciansFrom(currentDivision)
+            techniciansOnDivisionOptions = techniciansOnDivision.map(convertToOption)
+
+            const techniciansOnOtherDivisions =
+                differenceBy(
+                    uniqBy(
+                        otherDivisions.reduce((acc, otherDivision) => ([
+                            ...acc,
+                            ...getTechniciansFrom(otherDivision),
+                        ]), []),
+                        'id',
+                    ),
+                    techniciansOnDivision,
+                    'id',
+                )
+
+
+            techniciansOnOtherDivisionsOptions = techniciansOnOtherDivisions.map(convertToOption)
+
+            otherTechniciansOptions = differenceBy(employeeOptions, [
+                ...techniciansOnDivisionOptions,
+                ...techniciansOnOtherDivisionsOptions,
+            ], 'value')
+        }
+        const result = []
+        if (techniciansOnDivisionOptions.length > 0) {
+            result.push(
+                <Select.OptGroup label="Техники на участке">
```

Comment:
Move to translates

Reply:
Translated in `dd975bdc`.
Thank you for pointing on this.